src/Wizard.jl

```diff
@@ -0,0 +1,207 @@
+function yn_prompt(question, default = :y)
+    @assert default in (:y, :n)
+    while true
+        print(question, " ", default == :y ? "[Y/n]" : "[y/N]", ": ")
+        answer = lowercase(strip(readline()))
+        if isempty(answer)
+            return default
+        elseif answer == "y" || answer == "yes"
+            return :y
+        elseif answer == "n" || answer == "no"
+            return :n
+        else
+            println("Unrecognized answer. Answer `y` or `n`.")
+        end
+    end
+end
+
+function download_source(workspace, num)
+    println("Please enter a URL (git repository or tarball) to obtain the source code from.")
+    print("> ")
+    url = readline()
+    println()
+
+    source_path = joinpath(workspace, "source-$num.tar.gz")
+
+    download_cmd = gen_download_cmd(url, source_path)
+    oc = OutputCollector(download_cmd; verbose=true)
+    try
+        if !wait(oc)
+            error()
+        end
+    catch
+        error("Could not download $(url) to $(workspace)")
+    end
+
+    source_path
+end
+
+function run_wizard()
+    println("Welcome to the BinaryBuilder wizard.\n"*
+            "We'll get you set up in no time.\n")
+
+    print_with_color(:bold, "\t\t\t\# Step 1: Obtain the source code\n\n")
+
+    workspace = tempname()
+    mkpath(workspace)
+
+    source_files = String[]
+
+    num = 1
+    while true
+        push!(source_files, download_source(workspace, num))
+        println()
+        num += 1
+        yn_prompt("Would you like to download additional sources? ", :n) == :y || break
+    end
+
+    println()
+    print_with_color(:bold, "\t\t\t\# Step 2: Build for Linux x86_64\n\n")
+
+    println("You will now be dropped into the cross-compilation environment.")
+    println("Please compile the library. Your initial compilation target is Linux x86_64")
+    println("The \$DESTDIR environment variable contains the target directory.")
+    println("Many build systems will respect this variable automatically")
+    println("Once you are done, exit by typing `exit` or `^D`")
+
+    println()
+
+    build_path = tempname()
+    mkpath(build_path)
+    local history
+    cd(build_path) do
+        temp_prefix() do prefix
+            histfile = joinpath(build_path, ".bash_history")
+            dr = DockerRunner(prefix = prefix, platform = :linux64,
```

Comment:
Platform arguments should now be using `Platform` objects. In this case it should be `platform = Linux(:x86_64)`.

Reply:
True, there's a bunch of other code like this in this package also.

Reply:
Ah yeah you're right. This is Binary**Builder**, not Binary**Provider**. Perhaps I'll go through and update things here.

Reply:
I already have those changes locally, so if you haven't yet, don't start. Otherwise feel free to put up a PR.

Reply:
Okay, I just did. Didn't see your message until just now.